src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import './ContextMenu.scss';
+
+interface ContextMenuProps {
+  options: string[];
+  onSelect: (option: string) => void;
+}
+
+function ContextMenu({options, onSelect}: ContextMenuProps) {
+  const [menuOpened, setMenuOpened] = useState(false);
+
+  const handleToggleMenuVisibility = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.stopPropagation();
+    setMenuOpened(!menuOpened);
+  }
+
+  const handleOptionSelect = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>, option: string) => {
+    event.stopPropagation();
+    setMenuOpened(!menuOpened);
+    onSelect(option);
+  }
+
+  return (
+      <div className="position-relative">
+        <button aria-label="Options" className="context-menu__button" onClick={(event) => handleToggleMenuVisibility(event)}>
```

Comment:
There is no need in a anonymous wrapper you can simplify
```suggestion
        <button aria-label="Options" className="context-menu__button" onClick={handleToggleMenuVisibility}>
```